src/prisma/course.ts

```diff
@@ -119,70 +117,39 @@ async function upsertCourses(
 
   const prof = await prisma.user.findFirst({
     where: {
-      displayName: course.prof?.displayName,
+      // connect assignments for TBA to TBA user
+      displayName: course.prof?.displayName ?? 'TBA',
     },
   });
 
-  // Grab the title from previous entries
-  const existingCourse = await prisma.course.findFirst({
-    where: {
-      courseCode: course.courseNumber,
-      subject: course.subject,
-    },
-  });
+  if (!prof)
+    throw new Error('Unable to find professor to assign course sections to.');
 
-  await prisma.course.upsert({
-    create: {
-      courseCode: course.courseNumber,
-      title: existingCourse?.title ?? course.courseTitle,
-      subject: course.subject,
-      term: term,
-      streamSequence: utils.getSeqNumber(course.subject, course.courseNumber),
-      courseSection: {
-        create: {
-          sectionNumber: course.sequenceNumber,
-          capacity: course.courseCapacity ?? 0,
-          startDate,
-          endDate,
-          hoursPerWeek: 0,
-          schedule: { connect: { id: schedule.id } },
-          meetingTime: {
-            create: assignmentToMeetingTime(course.assignment),
-          },
-          user: {
-            connect: {
-              id: prof?.id,
-            },
-          },
+  await prisma.courseSection.create({
+    data: {
+      sectionNumber: course.sequenceNumber,
+      capacity: course.courseCapacity ?? 0,
+      startDate,
+      endDate,
+      hoursPerWeek: 0,
+      schedule: { connect: { id: schedule.id } },
+      meetingTime: {
+        create: assignmentToMeetingTime(course.assignment),
+      },
+      user: {
+        connect: {
+          id: prof.id,
         },
       },
-    },
-    update: {
-      courseSection: {
-        create: {
-          sectionNumber: course.sequenceNumber,
-          capacity: course.courseCapacity ?? 0,
-          startDate,
-          endDate,
-          hoursPerWeek: course.assignment?.hoursWeek ?? 0,
-          schedule: { connect: { id: schedule.id } },
-          meetingTime: {
-            create: assignmentToMeetingTime(course.assignment),
-          },
-          user: {
-            connect: {
-              id: prof?.id,
-            },
+      course: {
+        connect: {
+          subject_courseCode_term: {
+            courseCode: course.courseNumber,
+            subject: course.subject,
+            term: term,
```

Comment:
Would this make more sense as `connectOrCreate` in the case that there is no course with these parameters?

Reply:
We want to reject them as there shouldn't be unknown courses being scheduled at this point. The list of courses we use to seed to from frontend so we're matching in terms of courses we're dealing with now.

Reply:
Sounds good